### ISSUE TITLE
Passing RecyclableMemoryStream to StreamContent of HttpRequestMessage

### ISSUE THREAD

**Issue by megafinz** [microsoft/Microsoft.IO.RecyclableMemoryStream#117]
Yello.

I have a code somewhat like this:

```csharp
IHttpClient httpClient = …;
string address = …;
string fileName = …;
byte[] buffer = …;
CancellationToken ct = …;

var stream = RecyclableStreamManager.GetStream();
await stream.WriteAsync(buffer, 0, buffer.Length, ct);
stream.Seek(0, SeekOrigin.Begin);

using var content = new MultipartFormDataContent
{
    { new StreamContent(stream), "file", fileName }
};

using var request = new HttpRequestMessage
{
    RequestUri = new Uri(address),
    Method = HttpMethod.Post,
    Content = content
};

await httpClient.SendAsync(request, ct);
```

It doesn't work in a sense that server receives 0-length multipart body section. If I change `RecyclableStreamManager.GetStream()` to `new MemoryStream()`, everything works fine. 

What am I doing wrong? Is this task unsuitable for `RecyclableMemoryStream`?  Or maybe I need to configure something differently? Or is it a bug?

Thanks.

**Comment by benmwatson** [comment]
This should work just fine. I don't see an obvious issue in your code. What happens if you do `stream.Position = 0` instead? Could you  expand your example and create a complete repro case that I could run and take a look at?

**Comment by megafinz** [comment]
> Could you expand your example and create a complete repro case that I could run and take a look at?

Tried to do that and it seems that everything should be working fine. I am able to receive sample data correctly on the test server.

Then I changed `MemoryStream` in my actual code back to `RecyclableMemoryStream` and it also works! Not sure what magic is that, I don't remember changing anything else, but I definitely remember that it didn't work with `RecyclableMemoryStream` and changing it to `MemoryStream` fixed the problem.

Guess we can call it a fluke, sorry for bothering.
